src/main/java/org/broadinstitute/hellbender/engine/AssemblyRegionWalker.java

```diff
@@ -45,11 +48,8 @@
  */
 public abstract class AssemblyRegionWalker extends GATKTool {
 
-    @Argument(fullName="readShardSize", shortName="readShardSize", doc = "Maximum size of each read shard, in bases. For good performance, this should be much larger than the maximum assembly region size.", optional = true)
-    protected int readShardSize = defaultReadShardSize();
-
-    @Argument(fullName="readShardPadding", shortName="readShardPadding", doc = "Each read shard has this many bases of extra context on each side. Read shards must have as much or more padding than assembly regions.", optional = true)
-    protected int readShardPadding = defaultReadShardPadding();
+    @ArgumentCollection
+    protected ShardingArgumentCollection shardingArgumentCollection = defaultShardingArgumentCollection();
```

Comment:
This is the less disruptive way of doing this, but maybe will be useful to have abstract methods for getting size/step/padding and let tool developers decide if they expose them to the user or not. But that requires to remove the "default size/padding" getters and modify the tools implementing them.

I know that this is more complicated because one of the key tools in GATK will be affected by this: HaplotypeCaller. But  that is a good way to have finer control on what will be exposed to the final user, whenever the optimization for intervals is done.